toolkit/scripts/containerized-build/resources/setup_functions.sh

```diff
@@ -0,0 +1,146 @@
+#! /bin/bash
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+TOPDIR=<TOPDIR>
+SPECS_DIR=$TOPDIR/SPECS
+SOURCES_DIR=$TOPDIR/SOURCES
+RPMS_DIR=$TOPDIR/RPMS
+SRPMS_DIR=$TOPDIR/SRPMS
+IS_REPO_ENABLED=false
+
+# Mariner macro files used during spec parsing (as defined in toolkit/scripts/rpmops.sh)
+DEFINES=(-D "with_check 1")
+MACROS=()
+for macro_file in "$SPECS_DIR"/mariner-rpm-macros/macros* "$SPECS_DIR"/pyproject-rpm-macros/macros.pyproject "$SPECS_DIR"/perl/macros.perl
+do
+  MACROS+=("--load=$macro_file")
+done
+
+# Create symlink from SPECS/ to SOURCES/ when rpm is called
+rpm() {
+    local args=("$@")
+    command "$FUNCNAME" "${args[@]}"
+    if [[ ${args} = *"i"* ]]; then
+        for ((i = 0; i < ${#args[@]}; ++i)); do
+            if [[ ${args[$i]} = *".src.rpm"* ]]; then
+                local SPEC=${args[$i]}
+                SPEC=${SPEC##*/}
```

Comment:
Just curious - what do all these SPEC= assignments do exactly?

Reply:
It's just extracting the SPEC name from the .src.rpm filename, which will have version number, etc.

Reply:
My bash-fu is pretty rusty. How does the block work?

```
SPEC=${SPEC##*/}
SPEC=${SPEC%-*}
SPEC=${SPEC%-*}
SPEC=${SPEC%\**}
```

Is the repeated `SPEC=${SPEC%-*}` intentional?

Reply:
Yeah, it is extracting the correct spec name, because we have varied spec names. I don't remember the exact specs which were failing without this

Reply:
Bash can be a little bit of a symbol soup. It might be worth adding comments for what each symbol (e.g. `##`, `%`, `*`, etc.) is doing.

Reply:
Updated, I can't recall which SPEC needed the last one, and what it was accomplishing